tools/build_config/Jenkinsfile

```diff
@@ -1,118 +1,182 @@
+        if (isUnix()) {
+          sh """
+            curl -H "Authorization: token ${api_token}" \
+              --request POST \
+              --data '{"state": "${state}", \
+                "description": "${message}", \
+                "target_url": "$BUILD_URL", \
+                "context": "$JOB_BASE_NAME"}' \
+              $PR_STATUSES_URL > /dev/null
             """
         }
+        else {
+          powershell """
+            [Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"
+            \$payload = @{
+              "state" = "${state}";
+              "description" = "${message}";
+              "target_url" = "$BUILD_URL";
+              "context" = "$JOB_BASE_NAME"}
+
+            Invoke-RestMethod -URI "$PR_STATUSES_URL" \
+              -Headers @{Authorization = "token ${api_token}"} \
+              -Method 'POST' \
+              -Body (\$payload|ConvertTo-JSON) \
+              -ContentType "application/json"
+          """
+        }
+      }
     }
+  }
 }
 
 def base_job_name = "${env.JOB_BASE_NAME}".replace("PR-", "").replace("Branch-", "")
 
+/* Pipeline parameters:
+ *   All platforms:
+ *     AGENT (e.g. 'sl7', 'PACE Windows (Private)', 'osx')
+ *     MATLAB_VERSION (e.g R2018b)
+ *     RELEASE_TYPE ('nightly', 'release' - optional)
+ *   Unix only:
+ *     CMAKE_VERSION (e.g. 3.7.2)
+ *     GCC_VERSION (e.g. 6.3.0)
+ *   Windows only:
+ *     VS_VERSION (e.g. 2017)
+ */
 pipeline {
-    agent {
-        label env.AGENT
-    }
+  agent {
+    label env.AGENT
+  }
 
-    stages {
+  stages {
 
-        stage('Notify') {
-            steps {
-                post_github_status("pending", "The build is running")
-            }
-        }
+    stage('Notify') {
+      steps {
+        post_github_status("pending", "The build is running")
+      }
+    }
 
-        stage('Clone-Herbert') {
-            steps {
-                copyArtifacts(
-                    filter: 'build/Herbert-*',
-                    fingerprintArtifacts: true,
-                    projectName: "PACE-neutrons/Herbert/${base_job_name}",
-                    selector: lastSuccessful()
-                )
-                sh '''
-                    archive_name="\$(find -name Herbert-*.tar.gz)"
-                    tar --wildcards --strip-components=1 -xf \$archive_name */Herbert *.m
-                '''
-            }
+    stage('Clone-Herbert') {
+      steps {
+        copyArtifacts(
+          filter: 'build/Herbert-*',
+          fingerprintArtifacts: true,
+          projectName: "PACE-neutrons/Herbert/${base_job_name}",
+          selector: lastSuccessful()
+        )
+        script {
+          if (isUnix()) {
+            sh '''
+              archive_name="\$(find -name Herbert-*.tar.gz)"
+              tar --wildcards --strip-components=1 -xf \$archive_name */Herbert
+            '''
+          }
+          else {
+            powershell './tools/build_config/extract_herbert_artifact.ps1'
```

Comment:
same comment for all `powershell` blocks, these should be in an isWindows() guard.